src/main/java/org/carlmontrobotics/subsystems/IntakeShooter.java

```diff
@@ -4,47 +4,120 @@
-
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DigitalOutput;
 import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import edu.wpi.first.wpilibj2.command.WaitCommand; 
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkPIDController;
+
+import edu.wpi.first.math.controller.PIDController;
+import static org.carlmontrobotics.Constants.IntakeShooter.*;
+
+//import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+//import edu.wpi.first.networktables.NetworkTable;
+//import edu.wpi.first.networktables.NetworkTableEntry;
+//import edu.wpi.first.networktables.NetworkTableInstance;
+//these 4 imports above are for limelight
+
 
 
 public class IntakeShooter extends SubsystemBase {
-    CANSparkMax leftFiringMotor = MotorControllerFactory.createSparkMax(0, MotorConfig.NEO);
-    CANSparkMax rightFiringMotor = MotorControllerFactory.createSparkMax(1,MotorConfig.NEO);
-    CANSparkMax leftPassMotor = MotorControllerFactory.createSparkMax(2, MotorConfig.NEO_550);
-		CANSparkMax rightPassMotor = MotorControllerFactory.createSparkMax(3,MotorConfig.NEO_550);
-	
-		CANSparkMax intakeMotor = MotorControllerFactory.createSparkMax(4, MotorConfig.NEO_550);
-//	CANSparkMax rightIntakeMotor = MotorControllerFactory.createSparkMax(5, MotorConfig.NEO_550);
-	
-		public IntakeShooter() {
-    }
-	
-		public void intake() {}//run both intake and passing motors
-    public void fire() {}//run both passing and firing motors
+    CANSparkMax leftShootingMotor = MotorControllerFactory.createSparkMax(leftShootingPort, MotorConfig.NEO);
+    CANSparkMax rightShootingMotor = MotorControllerFactory.createSparkMax(rightShootingPort, MotorConfig.NEO);
+    CANSparkMax intakeMotor = MotorControllerFactory.createSparkMax(intakePort, MotorConfig.NEO_550);
+
+	private final RelativeEncoder leftShootingEncoder = leftShootingMotor.getEncoder();
+	private final RelativeEncoder rightShootingEncoder = rightShootingMotor.getEncoder();
+
+	private final RelativeEncoder intakeMotorEncoder = intakeMotor.getEncoder();
+	//todo: figure out way to make it so its not "constants."
+	private final SparkPIDController leftShootingMotorPID = leftShootingMotor.getPIDController();
+	private final SparkPIDController rightShootingMotorPID = rightShootingMotor.getPIDController();
+	private final TimeOfFlight distSensor = new TimeOfFlight(distSensorPort);
+
+	//private final DigitalOutput beamBreak = new DigitalOutput(beamBreakPort);
```

Comment:
we don't plan to use beam breaks. The effective distance is too short to be effective.